bookkeeper-server/src/test/java/org/apache/bookkeeper/client/BookKeeperTest.java

```diff
@@ -1127,4 +1132,87 @@ public void testBookieAddressResolverPassedToDNSToSwitchMapping() throws Excepti
         }
     }
 
+    @DataProvider
+    public static Object[][] IllegalLedgerCreationArguments() {
+        return new Object[][] {
+            {0, 0, 0},
+            {3, 3, -1},
+            {2, 3, 1},
+            {2, 2, 3},
+            {3, 3, 0}
+        };
+    }
+    @Test
+    @UseDataProvider("IllegalLedgerCreationArguments")
+    public void testCreateLedgerWithIllegalArguments(int ensSize, int writeQuorumSize, int ackQuorumSize) throws Exception {
+        final byte[] passwd = "testpasswd".getBytes();
+        final byte[] data = "data".getBytes();
+
+        try {
+            LedgerHandle lh = bkc.createLedger(ensSize, writeQuorumSize, ackQuorumSize,
+                BookKeeper.DigestType.CRC32, passwd);
+            for (int i = 0; i < 10; ++i) {
+                lh.addEntry(data);
+            }
+            lh.close();
+            fail();
+        } catch (Exception e) {
+            assertTrue(e instanceof BKException.BKIncorrectParameterException);
+            assertEquals(e.getMessage(), "Incorrect parameter input");
+        }
+
+        try {
+            LedgerHandle lh = bkc.createLedgerAdv(ensSize, writeQuorumSize, ackQuorumSize,
+                BookKeeper.DigestType.CRC32, passwd);
+            for (int i = 0; i < 10; ++i) {
```

Comment:
The dead code can be removed.

Reply:
We'd better not remove the addEntry code, because create Ledger will succeed and the addEntry will be blocked without this PR

Reply:
Make sense.